src/Chainweb/Rosetta/Utils.hs

```diff
@@ -42,6 +45,151 @@ import Chainweb.Version
+  toPairs (ContinuationMetaData curr next rk total) =
+    [ "current-step" .= toObject curr
+    , "first-step-request-key" .= rk
+    , "total-steps" .= total ]
+    <> omitNextIfMissing
+    where
+      omitNextIfMissing = case next of
+        Nothing -> []
+        Just ns -> [ "next-step" .= toObject ns ]
+  toObject contMeta = HM.fromList (toPairs contMeta)
+
+toContMeta :: ChainId -> P.PactExec -> ContinuationMetaData
+toContMeta cid pe = ContinuationMetaData
+  { _continuationMetaData_currStep = toContStep cid pe
+  , _continuationStep_nextStep = toContNextStep cid pe
+  , _continuationMetaData_pactIdReqKey = P._pePactId pe
+  , _continuationMetaData_totalSteps = P._peStepCount pe
+  }
+
+
+-- | Provides information on the step that was just executed.
+data ContinuationCurrStep = ContinuationCurrStep
+  { _continuationCurrStep_chainId :: !T.Text
+  -- ^ Chain id where step was executed
+  , _continuationCurrStep_stepId :: !Int
+  -- ^ Step that was executed or skipped
+  , _continuationCurrStep_rollbackAvailable :: Bool
+  -- ^ Track whether a current step allows for rollbacks
+  } deriving Show
+-- TODO: Add ability to detect if step was rolled back.
+-- TODO: document
+instance ToObject ContinuationCurrStep where
+  toPairs (ContinuationCurrStep cid step rollback) =
+    [ "chain-id" .= cid
+    , "step-id" .= step
+    , "rollback-available" .= rollback ]
+  toObject contCurrStep = HM.fromList (toPairs contCurrStep)
+
+toContStep :: ChainId -> P.PactExec -> ContinuationCurrStep
+toContStep cid pe = ContinuationCurrStep
+  { _continuationCurrStep_chainId = chainIdToText cid
+  , _continuationCurrStep_stepId = P._peStep pe
+  , _continuationCurrStep_rollbackAvailable = P._peStepHasRollback pe
+  }
+
+
+-- | Indicates if the next step of a continuation occurs in a
+--   different chain or in the same chain.
+newtype ContinuationNextStep = ContinuationNextStep
+  { _continuationNextStep_chainId :: T.Text
+  } deriving Show
+-- TODO: document
+instance ToObject ContinuationNextStep where
+  toPairs (ContinuationNextStep cid) = [ "target-chain-id" .= cid ]
+  toObject contNextStep = HM.fromList (toPairs contNextStep)
+
+-- | Determines if the continuation has a next step and, if so, provides
+--   the chain id of where this next step will need to occur.
+toContNextStep
+    :: ChainId
+    -> P.PactExec
+    -> Maybe ContinuationNextStep
+toContNextStep currChainId pe
+  | isLastStep = Nothing
+  -- TODO: Add check to see if curr step was rolled back.
+  --       This would also mean a next step is not occuring.
+  | otherwise = case P._peYield pe >>= P._yProvenance of
+      Nothing -> Just $ ContinuationNextStep $ chainIdToText currChainId
+      -- ^ next step occurs in the same chain
+      Just (P.Provenance nextChainId _) ->
+      -- ^ next step is a cross-chain step
+        Just $ ContinuationNextStep (P._chainId nextChainId)
+  where
+    isLastStep = succ $ P._peStep pe == P._peStepCount pe
```

Comment:
Does this need to be 

```haskell
isLastStep = (succ P._peStep pe) == P._peStepCount pe
```
?

Or does `==` bind more loosely than `succ`? Either way, should probably explicitly paren this.

Reply:
Followup: 

```haskell
П> succ False
True
П> succ True
*** Exception: Prelude.Enum.Bool.succ: bad argument
П> succ $ True == False
True
П> succ $ True == True
*** Exception: Prelude.Enum.Bool.succ: bad argument
```

Yeah, that looks like a potential bomb waiting to go off to me.

Reply:
You're right. I made it more explicit via:
```
nextStep = succ $ P._peStep pe
totalSteps = P._peStepCount pe
isLastStep = nextStep == totalSteps
```
I was getting a hlint warning when I added the explicit parems, but I think this solution makes it explicit and gets rid of that hlint warning.